Sources/Task.swift

```diff
@@ -518,11 +518,25 @@ extension Task {
 							}
 							group.leave()
 						}
-						
+
 						observer.send(value: .launch(self))
-						process.launch()
-						close(stdoutPipe.writeFD)
-						close(stderrPipe.writeFD)
+
+						if #available(macOS 10.13, *) {
+							do {
+								defer {
+									close(stdoutPipe.writeFD)
+									close(stderrPipe.writeFD)
+								}
```

Comment:
This `defer` could probably move outside the `if` to cover both cases?

Reply:
Is it okay the code block run after the following block? (it may be no problem)

```swift
lifetime += stdinProducer.start()

lifetime.observeEnded {
	process.terminate()
}
```

Reply:
I don't think that'd be a problem?